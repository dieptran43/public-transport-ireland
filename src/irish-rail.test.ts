'use strict'

import { join } from 'path'
import { readFileSync } from 'fs'
import nock from 'nock'
import irishRail from './irish-rail'

const wsdlResponse = readFileSync(join(__dirname, 'fixtures', 'irish-rail', 'wsdl.xml'))
const allStationsResponseData = readFileSync(join(__dirname, 'fixtures', 'irish-rail', 'get-all-stations.xml'))

// mock WSDL request
nock('http://api.irishrail.ie')
  .get('/realtime/realtime.asmx?WSDL')
  .reply(200, wsdlResponse)

// mock get all stations request
nock('http://api.irishrail.ie', {
  reqheaders: {
    SOAPAction: '"http://api.irishrail.ie/realtime/getAllStationsXML_WithStationType"'
  }
})
  .post('/realtime/realtime.asmx')
  .once()
  .reply(200, allStationsResponseData)

test('It should get all stations', async function () {
  const stations = await irishRail.getStations()
  expect(stations).toMatchSnapshot()
})
