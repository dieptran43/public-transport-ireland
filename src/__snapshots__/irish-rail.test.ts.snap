// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should get all stations 1`] = `
Array [
  Object {
    "code": "BFSTC",
    "id": 228,
    "latitude": 54.6123,
    "longitude": -5.91744,
    "name": "Belfast",
  },
  Object {
    "code": "LBURN",
    "id": 238,
    "latitude": 54.514,
    "longitude": -6.04327,
    "name": "Lisburn",
  },
  Object {
    "code": "LURGN",
    "id": 241,
    "latitude": 54.4672,
    "longitude": -6.33547,
    "name": "Lurgan",
  },
]
`;
